File: frontend/src/api/market.js
// 市场洞察相关 API, 全部走 /api 代理(5173 -> 8080)
// 注意: 响应拦截器返回的是完整 { code, data } 信封, 这里统一拆出 data,
// 使 store 拿到的就是业务数组/对象(与 store 内部的数组校验一致)。
import request from './request'

const market = {
  kpi:         (q) => request.get('/market/kpi', { params: q }).then(r => r.data),
  trend:       (q) => request.get('/market/trend', { params: q }).then(r => r.data),
  competitors: (q) => request.get('/market/competitors', { params: q }).then(r => r.data),
  regions:     (q) => request.get('/market/regions', { params: q }).then(r => r.data),
  prices:      (q) => request.get('/market/prices', { params: q }).then(r => r.data),
  list:        (q) => request.get('/market/list', { params: q }).then(r => r.data)
}

export default market

File: backend/internal/api/handler/market.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"insta360-insight/internal/model"
	"insta360-insight/internal/service"
)

type Market struct{ svc *service.MarketService }

func NewMarket(svc *service.MarketService) *Market { return &Market{svc: svc} }

func (h *Market) Kpi(c *gin.Context) {
	d, e := h.svc.Kpi(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Market) Trend(c *gin.Context) {
	d, e := h.svc.Trend(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Market) Competitors(c *gin.Context) {
	d, e := h.svc.Competitors(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Market) Regions(c *gin.Context) {
	d, e := h.svc.Regions(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Market) Prices(c *gin.Context) {
	d, e := h.svc.Prices(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Market) List(c *gin.Context) {
	d, e := h.svc.List(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
